openrouteservice/src/main/java/heigit/ors/routing/RoutingProfile.java

```diff
@@ -761,16 +761,18 @@ else if (bearings[1] == null)
             if (_astarApproximation != null)
                 req.getHints().put("astarbi.approximation", _astarApproximation);
 
-			/*if (directedSegment)
-				resp = mGraphHopper.directRoute(req); NOTE IMPLEMENTED!!!
-			else */
-
-            mGraphHopper.setSimplifyResponse(geometrySimplify);
-            resp = mGraphHopper.route(req);
-
-            if (DebugUtility.isDebug()) {
+            if (directedSegment) {
+                resp = mGraphHopper.directRoute(req);
+            } else {
+                mGraphHopper.setSimplifyResponse(geometrySimplify);
+                resp = mGraphHopper.route(req);
+            }
+            if (DebugUtility.isDebug() && !directedSegment) {
                 System.out.println("visited_nodes.average - " + resp.getHints().get("visited_nodes.average", ""));
             }
+            if (DebugUtility.isDebug() && directedSegment) {
+                System.out.println("skipped segment - " + resp.getHints().get("skipped_segment", ""));
```

Comment:
Should use the LOGGER here rather than System.out.println

Reply:
Should I also adjust the old println calls above?

Reply:
can do...